src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericTable.js

```diff
@@ -21,11 +21,17 @@ const clone = rfdc();
 const DEFAULT_DATE_FORMAT = 'yyyy-MM-dd';
 const MAX_ERRORS_COUNT = 100;
 
+export const _getColumnWithoutDepths = (columns) => {
+  return columns.flatMap((columnOrColumnGroup) => {
+    return columnOrColumnGroup.children ? _getColumnWithoutDepths(columnOrColumnGroup.children) : columnOrColumnGroup;
+  });
+};
+
```

Comment:
I think the plural is not necessary in the function name. 

Also, why did you decide not to use `flat(Infinity)` method as you mentioned before? Would it be simpler to flatten the whole array instead of doing it element by element?

Reply:
I modified this with Nicolas, which is better syntaxe, less macro used and safer for array parameters in a field.